### PR TITLE
Add plugin dev doc to TOC

### PR DESCRIPTION
**Pull request by mstanleyjones**
### Describe the proposed changes

Add link to the TOC
While I'm at it, test and update the content for clarity (several commands needed sudo, for instance)

### Project version

n/a

### Related issue

#252

### Please take a look

@icecrime @vieux @thaJeztah

### CODE REVIEW

File: engine/extend/index.md [hunk 1]
```diff
@@ -17,39 +13,33 @@ title: Managed plugin system
...
+## Installing and using a plugin
 
-For information on Docker Engine plugins generally available in Docker Engine
-1.12 and earlier, refer to [Understand legacy Docker Engine plugins](legacy_plugins.md).
+Plugins are distributed as Docker images and can be hosted on Docker Hub or on
+a private registry.
 
-## How to operate a plugin
+To install a plugin, use the `docker plugin install` command, which pulls the
+plugin from the the configured registry, prompts you to grant permissions if
```
Comment (thaJeztah) [inline engine/extend/index.md]: s/the the/the/

not sure about "configured registry", as the plugins reference determines which registry to use (e.g. `myregistry:5000/my/plugin:latest`). Perhaps just "registry" (or "Docker Hub")

File: engine/extend/index.md [hunk 2]
```diff
@@ -87,39 +80,39 @@ started without error.
...
-    $ docker run -v sshvolume:/data busybox ls /data
+    $ docker volume ls
 
-    <content of /remote on machine 1.2.3.4>
+    DRIVER              NAME
+    vieux/sshfs         sshvolume
     ```
 
-5.  Verify the plugin successfully created the volume.
+5.  Use the volume `sshvolume` to start a container.
```
Comment (thaJeztah) [inline engine/extend/index.md]: Almost reads as "the volume starts the container" :rabbit:

File: engine/extend/index.md [hunk 3]
```diff
@@ -246,26 +253,40 @@ with the Docker Engine.
...
+1. Choose a name for the plugin. Plugin name uses the same format as images,
+   for example: `<repo_name>/<name>`.
 
-0. Choose a name for the plugin. Plugin name uses the same format as images,
-for example: `<repo_name>/<name>`.
-1. Create a rootfs in `/var/lib/docker/plugins/$id/rootfs`.
-2. Create manifest.json file in `/var/lib/docker/plugins/$id/`.
-3. Create a `plugin-config.json` if needed.
-4. Create or add a section to `/var/lib/docker/plugins/plugins.json`. Use
+2. Create a `rootfs` and export it to `/var/lib/docker/plugins/$id/rootfs`.
```
Comment (thaJeztah) [inline engine/extend/index.md]: Not sure what is meant by "export it to" here

File: engine/extend/index.md (hunk 1 above)
Comment (mstanleyjones) [inline engine/extend/index.md]: Done

File: engine/extend/index.md (hunk 2 above)
Comment (mstanleyjones) [inline engine/extend/index.md]: Yeah I misunderstood this the first time around. Fixed.

File: engine/extend/index.md (hunk 3 above)
Comment (mstanleyjones) [inline engine/extend/index.md]: Added some explanation and a pointer to the example higher up in the file.

**Comment by johndmulhausen** [comment]
Looks good from a docs perspective; once @thaJeztah 's comments are addressed, merge away!

File: engine/extend/index.md [hunk 4]
```diff
@@ -78,6 +69,8 @@ started without error.
     ```
 
 3.  Create a volume using the plugin.
+    This example mounts the `/remmote` directory on host `1.2.3.4` into a
```
Comment (thaJeztah) [inline engine/extend/index.md]: typo; `remmote`

File: engine/extend/index.md [hunk 5]
```diff
@@ -78,6 +69,8 @@ started without error.
...
+    volume nammed `sshvolume`. This volume can now be mounted into containers.
```
Comment (thaJeztah) [inline engine/extend/index.md]: typo `nammed`

**Comment by johndmulhausen** [comment]
Can I ask why /engine/extend/ wouldn't live in the docs repo? It is clearly a handwritten tutorial and not something that would ever be auto-generated, and not any more release-bound than any other writer-curated content that could be tracked in `vnext-engine`

https://docker.github.io/engine/extend/

**Comment by thaJeztah** [comment]
@johndmulhausen because a large part of it is "reference", which is tightly coupled to code changes, also there's a manifest in there, that should not be separated from the code.

We may consider moving those parts underneath "reference" (plugin-api, plugin manifest?)

**Comment by mstanleyjones** [comment]
Moved the rewrite section of the PR to https://github.com/docker/docker/pull/27593

**Comment by johndmulhausen** [comment]
@thaJeztah Fair enough. Currently, NOT_EDITED_HERE.txt is marking all of /engine/extend/ as "not edited here" -- is it just index.md, or is it really the whole directory?

**Comment by mstanleyjones** [comment]
I'll split this up and only keep the menu part here, move the edits to
docker/docker. Thanks for the reminder. I can't wait for this to be
automated!

**Pull request merged**